test/provider.js

```diff
@@ -0,0 +1,33 @@
+const Ganache = require('ganache-core');
+const assert = require('assert');
+const WalletProvider = require('../index.js');
+const mnemonic = 'candy maple cake sugar pudding cream honey rich smooth crumble sweet treat';
+
+describe("HD Wallet Provider", function(done) {
+  var Web3 = require('web3');
+  var web3 = new Web3();
+  var port = 8545;
+  var server;
+  var provider;
+
+  before(done => {
+    server = Ganache.server();
+    server.listen(port, done);
+  });
+
+  after(done => {
+    provider.engine.stop();
+    setTimeout(() => server.close(done), 2000); // :/
+  })
```

Comment:
If we don't do this `eth-block-tracker` continues to poll in the background after the server closes and 
+  mocha won't close
+  we get `Invalid JSON RPC response`.

Possible this is the underlying problem with [truffle 852](https://github.com/trufflesuite/truffle/issues/852).